addon/ActivityStreams.js

```diff
@@ -239,7 +239,7 @@ ActivityStreams.prototype = {
     this._store.dispatch(am.actions.Response("PREFS_RESPONSE", simplePrefs.prefs));
 
     // Share
-    this._store.dispatch(am.actions.Response("SHARE_PROVIDERS_RESPONSE", this._shareProvider.socialProviders));
+    this._store.dispatch(am.actions.Response("SHARE_PROVIDERS_RESPONSE", this._shareProvider.socialProviders || []));
```

Comment:
There is a *really* weird bug in the tests in which it somehow returns `undefined`, which should be impossible given the code path... the only thing i can think of is that it's got a reference to an object that isn't in the right scope or something. It appears to be a race condition related to the `Social` service. I filed https://github.com/mozilla/activity-stream/issues/1314 as a follow-up, since @rlr knows that code better than me